core/services/ocr2/plugins/ocr2vrf/coordinator/coordinator.go

```diff
@@ -73,11 +93,10 @@ type coordinator struct {
 	evmClient evmclient.Client
 
 	// set of blocks that have been scheduled for transmission.
-	toBeTransmittedBlocks map[block]struct{}
+	toBeTransmittedBlocks *blockCache[blockInReport]
 	// set of request id's that have been scheduled for transmission.
-	toBeTransmittedCallbacks map[callback]struct{}
-	// transmittedMu protects the toBeTransmittedBlocks and toBeTransmittedCallbacks
-	transmittedMu sync.Mutex
+	toBeTransmittedCallbacks *blockCache[callbackInReport]
+	transmittedMu            sync.Mutex
```

Comment:
Might be worth checking if this is still needed, could simplify this component.

Reply:
Actually, it turns out `confirmationDelays` is being passed directly into ReportBlocks, and the item being fed to it is `s.confirmationDelays`, so that's probably a concern. I think we should leave the mutex and in another PR make the use of `confirmationDelays` more thread-safe.

Reply:
That's a good observation - reading from slices is safe from multiple goroutines (see https://groups.google.com/g/golang-nuts/c/nL8z96SXcDs?pli=1) though in this case that doesn't really apply, since `confirmationDelays` is only read from the `filter*` methods which do not get called concurrently with each other, so should be safe to remove this mutex.